lib/data/external/AwsClient.js

```diff
@@ -117,18 +124,25 @@ class AwsClient {
         return callback(null, stream);
     }
     delete(objectGetInfo, reqUids, callback) {
-        // for backwards compatibility
-        const key = typeof objectGetInfo === 'string' ? objectGetInfo :
-          objectGetInfo.key;
+        const { key, dataStoreVersionId, deleteVersion } = objectGetInfo;
```

Comment:
no need for backwards compat anymore?

Reply:
Nope because if I understand correctly, if we have the object stored in metadata as existing on the AWS backend, that object must have been put with the new format. (Couldn't have put the object with the old format on AWS backend since AWS backend didn't exist when we were still using the old format)